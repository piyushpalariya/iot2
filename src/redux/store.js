import {configureStore} from "@reduxjs/toolkit"
import LanguageReducer from "./LanguageSlice"
import isLoggedInReducer from "./LoggedSlice"

export const store = configureStore({
  reducer: {
    Language: LanguageReducer,
    isLoggedIn: isLoggedInReducer,
  },
})

export default store
