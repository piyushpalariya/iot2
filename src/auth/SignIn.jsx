import { Text, View, TextInput } from "react-native";
import React, { useState } from "react";
import CustomButton from "../components/CustomButton";
// import {useNavigation} from "@react-navigation/native"
// import translate from "../locale/languages"
import { /*useSelector,*/ useDispatch } from "react-redux";
import { setIsLoggedIn } from "../redux/LoggedSlice";
import handleBiometricAuth from "./Biometrics";
import { storage } from "../components/MMKV";
import { z } from "zod";

const loginSchema = z.object({
  // email: z
  //   .string()
  //   .min(3, "Username must be at least 3 characters")
  //   .max(50, "Username must be at most 50 characters"), // Example constraints
  password: z
    .string()
    .min(2, "Password must be at least 8 characters")
    .max(100, "Password must be at most 100 characters"),
  email: z.string().email("Invalid email format"),
});

const ValidateLogin = (credentials) => {
  try {
    const result = loginSchema.parse(credentials);
    console.log("Login successful:", result);
  } catch (error) {
    return JSON.parse(error);
  }
};

export default function SignIn() {
  const [email, setEmail] = useState(null);
  const [password, setPassword] = useState(null);
  const [Error, setError] = useState("");

  const dispatch = useDispatch();
  const LoginWithBiometrics = async () => {
    const Authenticate = await handleBiometricAuth();
    if (Authenticate) {
      dispatch(setIsLoggedIn(true));
    }
  };
  // const Language = useSelector(state => state.Language.LanguageName)
  const BiometricsEnabled = storage.getBoolean("BiometricsEnabled");
  return (
    <View className="bg-[#f7f5f5] w-full h-full  justify-start items-center dark:bg-[#0a0a0a]">
      <Text className="mt-20 text-2xl" testID="page title">
        LOGIN
      </Text>
      <View className="bg-[#d1d8de] mt-16 w-[75%] h-[40%] justify-start items-center rounded-[25]">
        <TextInput
          className="mt-20"
          testID="email"
          placeholder="email"
          value={email}
          onChangeText={setEmail}
        />
        {Error && <Text>{Error?.[0]?.message || ""}</Text>}
        <TextInput
          testID="password"
          placeholder="password"
          value={password}
          onChangeText={setPassword}
        />
        {Error && <Text>{Error?.[1]?.message || ""}</Text>}
        <CustomButton
          className="mt-8 bg-yellow-500 rounded-[25] m-1 p-3  items-center justify-center"
          testID="login button"
          onPress={() => {
            dispatch(setIsLoggedIn(true));
            const result = ValidateLogin({ email, password });
            setError(result);
          }}
          title="Login"
        />
      </View>

      {BiometricsEnabled && (
        <CustomButton
          testID="login with Biometrics"
          onPress={LoginWithBiometrics}
          title="Login with Biometrics"
        />
      )}
    </View>
  );
}
