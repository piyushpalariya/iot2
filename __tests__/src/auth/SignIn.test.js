import SignIn from "../../../src/auth/SignIn"
import {render, screen, userEvent} from "@testing-library/react-native"
import React from "react"
import {Provider} from "react-redux"
import store from "../../../src/redux/store"
import {NavigationContainer} from "@react-navigation/native"

const user = userEvent.setup()

describe("SignIn", () => {
  it("snapshot", () => {
    const {toJSON} = render(
      <Provider store={store}>
        <NavigationContainer>
          <SignIn />
        </NavigationContainer>
      </Provider>,
    )
    expect(toJSON()).toMatchSnapshot()
  })
  it("renders correctly", () => {
    render(
      <Provider store={store}>
        <NavigationContainer>
          <SignIn />
        </NavigationContainer>
      </Provider>,
    )
    expect(screen.getByTestId("page title")).toBeOnTheScreen()
    expect(screen.getByTestId("email")).toBeOnTheScreen()
    expect(screen.getByTestId("password")).toBeOnTheScreen()
    expect(screen.getByTestId("login button")).toBeOnTheScreen()
  })
  it("should have correct title", () => {
    render(
      <Provider store={store}>
        <NavigationContainer>
          <SignIn />
        </NavigationContainer>
      </Provider>,
    )
    expect(screen.getByTestId("page title")).toHaveTextContent("Signin")
  })

  it("should have correct placeholders", () => {
    render(
      <Provider store={store}>
        <NavigationContainer>
          <SignIn />
        </NavigationContainer>
      </Provider>,
    )
    expect(screen.getByTestId("email")).toHaveProp("placeholder", "email")
    expect(screen.getByTestId("password")).toHaveProp("placeholder", "password")
  })
  it("button should be pressable ", () => {
    render(
      <Provider store={store}>
        <NavigationContainer>
          <SignIn />
        </NavigationContainer>
      </Provider>,
    )
    const SignInButton = screen.getByTestId("login button")
    user.press(SignInButton)
  })
})
