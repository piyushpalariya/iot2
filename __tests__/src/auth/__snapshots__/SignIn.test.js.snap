// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn snapshot 1`] = `
<View
  className="bg-[#f7f5f5] w-full h-full  justify-start items-center dark:bg-[#0a0a0a]"
>
  <Text
    className="mt-20 text-2xl"
    testID="page title"
  >
    LOGIN
  </Text>
  <View
    className="bg-[#d1d8de] mt-16 w-[75%] h-[40%] justify-start items-center rounded-[25]"
  >
    <TextInput
      className="mt-20"
      onChangeText={[Function]}
      placeholder="email"
      testID="email"
      value={null}
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="password"
      testID="password"
      value={null}
    />
    <View
      className="mt-8 bg-yellow-500 rounded-[25] m-1 p-3  items-center justify-center"
      testID="login button"
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          className="text-center"
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</View>
`;
